specs/protocol/holocene/derivation.md

```diff
@@ -1,13 +1,332 @@
+during sync start yet._
+
+# Rationale
+
+## Strict Frame and Batch Ordering
+
+Strict Frame and Batch Ordering simplifies implementations of the derivation pipeline, and leads to
+better worst-case cached data usage.
+
+- The frame queue only ever holds frames from a single batcher transaction.
+- The channel bank only ever holds a single staging channel, that is either being built up by
+incoming frames, or is is being processed by later stages.
+- The batch queue only ever holds at most a single span batch (that is being processed) and a single singular
+batch (from the span batch, or the staging channel directly)
+- The sync start greatly simplifies in the average production case.
+
+This has advantages for Fault Proof program implementations.
+
+## Partial Span Batch Validity
+
+Partial Span Batch Validity guarantees that a valid singular batch derived from a span batch can
+immediately be processed as valid and advance the safe chain, instead of being in an undecided state
+until the full span batch is converted into singular batches. This leads to swifter derivation and
+gives strong worst-case guarantees for Fault Proofs because the vailidty of a block doesn't depend
+on the validity of any future blocks any more. Note that before Holocene, to verify the first block
+of a span batch required validating the full span batch.
+
+## Fast Channel Invalidation
+
+The new Fast Channel Invalidation rule is a consistency implication of the Strict Ordering Rules.
+Because batches inside channels must be ordered and contiguous, assuming that all batches inside a
+channel are self-consistent (i.e., parent L2 hashes point to the block resulting from the previous
+batch), an invalid batch also forward-invalidates all remaining batches of the same channel.
+
+## Steady Block Derivation
+
+Steady Block Derivation changes the derivation rules for invalid payload attributes, replacing an
+invalid payload by a deposit-only/empty payload. Crucially, this means that the effect of an invalid
+payload doesn't propagate backwards in the derivation pipeline. This has benefits for Fault Proofs
+and Interop, because it guarantees that batch validity is not influenced by future stages and the
+block derived from a valid batch will be determined by the engine stage before it pulls new payload
+attributes from the previous stage. This avoids larger derivation pipeline resets.
+
+## Less Defensive Protocol
+
+The stricter derivation rules lead to a less defensive protocol. The old protocol rules allowed for
+second chances for invalid payloads and submitting frames and batches within channels out of order.
+Experiences from running OP Stack chains for over one and a half years have shown that these relaxed
+derivation rules are (almost) never needed, so stricter rules that improve worst-case scenarios for
+Fault Proofs and Interop are favorable.
+
+# Security and Implementation Considerations
```

Comment:
Similarly, this material could end up in the FMA. Do we also want it in the specs?

Reply:
Yes, we want to have a brief security discussion here, but not of the extend as the FMA. E.g. if you look at `fjord/derivation.md`, each feature has its own "Rationale" and "Security Considertaions" section.